### PR TITLE
Prepare for parallel read safety

### PR DESCRIPTION
**Pull request by ViktorHaag**
Use builtin python domain as pattern for parallel read safety:

- revise clear_doc() to be in line with python domain's clear_doc()
- add merge_domain() built in the same mode as python domain's merge_domain()

### CODE REVIEW

**Comment by ViktorHaag** [comment]
addresses issue #19

File: sphinxcontrib/phpdomain.py [hunk 1]
```diff
@@ -604,13 +605,21 @@ class PhpDomain(Domain):
...
+        for fullname, (fn, _l) in list(self.data['objects'].items()):
             if fn == docname:
                 del self.data['objects'][fullname]
-        for ns, (fn, _, _) in self.data['namespaces'].items():
+        for ns, (fn, _x, _x) in list(self.data['namespaces'].items()):
             if fn == docname:
                 del self.data['namespaces'][ns]
 
+    def merge_domaindata(self, docnames, otherdata):
+        for fullname, (fn, objtype) in otherdata['objects'].items():
```
Comment (markstory) [inline sphinxcontrib/phpdomain.py]: Why don't you need a list() cast on this items() but you do need it on the items() calls on 608 and 611?

Comment (ViktorHaag) [reply sphinxcontrib/phpdomain.py]: To be honest, I don't know. I'm happy to remove the changes to clear_doc(); the only reason I made those changes was to have both methods in-line with the python domain's code, as that seemed safe. Let me know if you'd like me to go back to the way clear_doc() was, and apologies for not being more clear on why the changes got proposed the way they did.

Comment (markstory) [reply sphinxcontrib/phpdomain.py]: I'm not well steeped in py3 tricks so I didn't know if those were load bearing list() casts.

Comment (ViktorHaag) [reply sphinxcontrib/phpdomain.py]: This [Stack answer](https://stackoverflow.com/a/3294899/608191) seems to suggest that, in Py3, dict.items() returns an iterator which can be affected by changes to dict during iteration over the list, while in Py2, dict.items() returns a list of `(key,value)` which _won't_ change to reflect any change to the underlying dict's items that happen during iteration over the list.

If what you want is to force Py3 to freeze the contents of the items list to what it was at the point of calling .items(), then the pattern is to do `list(d.items())` and then iterate over that list.

Which of these two scenarios is appropriate in these cases, I am not entirely sure. I just figured that the maintainers of the python domain code knew what they were doing and that this wasn't an oversight on their part. That assumption may not be warranted. 8/

**Pull request merged**